e2e/tests/node_not_ready/node_power_off.robot

```diff
@@ -0,0 +1,550 @@
+*** Settings ***
+Documentation       Test the Longhorn resillence if cluster node powering off
+
+Resource            ../../keywords/common.resource
+Resource            ../../keywords/engine.resource
+Resource            ../../keywords/replica.resource
+Resource            ../../keywords/node.resource
+Resource            ../../keywords/volume.resource
+
+Suite Setup         set_test_suite_environment
+Test Setup          set_test_environment    ${TEST NAME}
+Test Teardown       Cleanup resource and resume state
+
+
+*** Variables ***
+${Gi}=                  2**30
+${volume_size_gb}=      1
+${sleep_interval}=      300
+${volume_type}=         RWO
+
+
+*** Test Cases ***
+Node power off with replica-VA
+    [Documentation]    | =node 1= | =node 2= | =node 3= |
+    ...    | replica | replica | replica |
+    ...    | attached | | |
+    ...    | *power off* | | |
+    ${number_of_replicas}=    Convert To Integer    3
+    Set Test Variable    ${number_of_replicas}
+
+    ${volume_type}=    Evaluate    "${volume_type}".lower()
+
+    ${field1}=    Convert To String    {"spec": {"size": "${${volume_size_gb} * ${Gi}}"}}
+    ${field2}=    Convert To String    {"spec": {"numberOfReplicas": ${number_of_replicas}}}
+    ${field3}=    Convert To String    {"spec": {"accessMode": "${volume_type}"}}
+    @{list_of_fields}=    Create List    ${field1}    ${field2}    ${field3}
+
+    Given Create Volume With Fields    ${list_of_fields}
+    And Attach volume to node 1
+    And Write data into mount point
+
+    When Power off node 1
+
+    Then Node 1 should have 1 volume replica
```

Comment:
For these keywords, what will happen if there are more than 1 volume involved in this test case? Will it work? If not, how can we modify to make it work?

Manipulating more than 1 volume in a single test case is possible like in: https://github.com/longhorn/longhorn-tests/blob/master/manager/integration/tests/test_settings.py#L723

We should have a clearer idea before we go on and implement more details.

Reply:
The current design method is indeed only for one test volume in a case, and for the situation that multiple test volumes exist at the same time, I will open a ticket for further discussion later.

Reply:
https://github.com/longhorn/longhorn/issues/6498